src/views/ApartmentApp/components/StaffMenu/index.js

```diff
@@ -0,0 +1,174 @@
+import React, { useState, useEffect, useCallback } from 'react';
+import { CSVLink } from 'react-csv';
+import { Grid, Card, CardHeader, CardContent, Button, Typography } from '@material-ui/core/';
+import GetAppIcon from '@material-ui/icons/GetApp';
+import { DateTime } from 'luxon';
+import GordonLoader from '../../../../components/Loader';
+import housing from '../../../../services/housing';
+import ApplicationsTable from './components/ApplicationTable';
+
+/**
+ * @typedef { import('../../../../services/housing').ApplicationDetails } ApplicationDetails
+ * @typedef { import('../../../../services/housing').ApartmentApplicant } ApartmentApplicant
+ * @typedef { import('../../../../services/housing').ApartmentChoice } ApartmentChoice
+ */
+
+/**
+ * Renders the page for the apartment application housing staff menu
+ * @param {Object} props The React component props
+ * @param {*} props.authentication The user authentication
+ * @param {StudentProfileInfo} props.userProfile The student profile info of the current user
+ * @returns {JSX.Element} JSX Element for the staff menu web page
+ */
+const StaffMenu = ({ userProfile, authentication }) => {
+  const [loading, setLoading] = useState(true);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} ApplicationDetails */
+  const [applications, setApplications] = useState([]);
+
+  /** @type {[ApplicationDetails[], React.Dispatch<React.SetStateAction<ApplicationDetails[]>>]} Array of application details, after being formatted for use with react-csv */
+  const [applicationJsonArray, setApplicationJsonArray] = useState([]);
+
+  /** @type {[ApartmentApplicant[], React.Dispatch<React.SetStateAction<ApartmentApplicant[]>>]} Array of applicant info, after being formatted for use with react-csv */
+  const [applicantJsonArray, setApplicantJsonArray] = useState([]);
+
+  /** @type {[ApartmentChoice[], React.Dispatch<React.SetStateAction<ApartmentChoice[]>>]} Array of apartment choice info, after being formatted for use with react-csv */
+  const [apartmentChoiceJsonArray, setApartmentChoiceJsonArray] = useState([]);
+
+  const [dateStr, setDateStr] = useState('');
+  const filePrefix = 'apartapp';
+
+  /**
+   * Attempt to load all existing applications for the current semester
+   */
+  const loadAllCurrentApplications = useCallback(async () => {
+    setLoading(true);
+    let applicationDetailsArray = await housing.getAllApartmentApplications();
+    setApplications(applicationDetailsArray ?? []);
+    setLoading(false);
+  }, []);
```

Comment:
This is only called in the below `useEffect`, and it has 0 external dependencies, so I don't see a reason for it to be wrapped with `useCallback`. I think it would be much simpler to declare it within the `useEffect`.

Reply:
For this one, I think the idea was to have a button that could be clicked to reload the applications, but I think that should be equally accomplished by just refreshing the page. In that case, there is no reason to have it be separate like this

Reply:
That actually makes a lot of sense, and would be worthwhile to keep I think! Refreshing the page does a LOT of extra work becuase the ENTIRE component tree has to be recreated from scratch. Having a button to only refresh the data we're actually concerned about being stale would be much better user experience I think.

Reply:
I think this was the same reason for doing that on the student page as well, since the act of changing the application editor should force it to save the application and then load the new data, to make sure it is displaying the latest info while also preventing the a non-editor from attempting to make changes 

Additionally, if the frontend changes the editor state but that action fails in the API, then the page would make it appear as if it succeeded by disabling the interactive parts of the application, when in reality the user is still the editor in the backend/database. This could be handled with other forms of try/catch stuff around the API request to change the editor, but I think it is best an most reliable to make the page reload the application data so that it ensures the data displays to the user is an accurate reflection of what is actually in the database